fury/actor.py

```diff
@@ -1440,42 +1446,92 @@ def dots(points, color=(1, 0, 0), opacity=1, dot_size=5):
     :func:`fury.actor.point`
 
     """
-    if points.ndim == 2:
-        points_no = points.shape[0]
-    else:
-        points_no = 1
-
-    polyVertexPoints = Points()
-    polyVertexPoints.SetNumberOfPoints(points_no)
-    aPolyVertex = PolyVertex()
-    aPolyVertex.GetPointIds().SetNumberOfIds(points_no)
-
-    cnt = 0
-    if points.ndim > 1:
-        for point in points:
-            polyVertexPoints.InsertPoint(cnt, point[0], point[1], point[2])
-            aPolyVertex.GetPointIds().SetId(cnt, cnt)
-            cnt += 1
-    else:
-        polyVertexPoints.InsertPoint(cnt, points[0], points[1], points[2])
-        aPolyVertex.GetPointIds().SetId(cnt, cnt)
-        cnt += 1
+    if points.ndim != 2:
+        raise ValueError('Invalid points. The shape of the structure must be '
+                         '(Nx3). Your data has {} dimensions.'
+                         .format(points.ndim))
+
+    if points.shape[1] != 3:
+        raise ValueError('Invalid points. The shape of the last dimension '
+                         'must be 3. Your data has a last dimension of {}.'
+                         .format(points.shape[1]))
+
+    vtk_vertices = Points()
+    vtk_faces = CellArray()
+
+    # Add points
+    for i in range(len(points)):
+        p = points[i]
+        idd = vtk_vertices.InsertNextPoint(p)
+        vtk_faces.InsertNextCell(1)
+        vtk_faces.InsertCellPoint(idd)
+
+    color_tuple = __color_check(points, colors)
+    color_array, global_opacity = color_tuple
+
+    # Create a polydata object
+    polydata = PolyData()
+    polydata.SetPoints(vtk_vertices)
+    polydata.SetVerts(vtk_faces)
+    polydata.GetPointData().SetScalars(color_array)
+
+    # Visualize
+    mapper = PolyDataMapper()
+    mapper.SetInputData(polydata)
+
+    # Create an actor
+    poly_actor = Actor()
+    poly_actor.SetMapper(mapper)
 
-    aPolyVertexGrid = UnstructuredGrid()
-    aPolyVertexGrid.Allocate(1, 1)
-    aPolyVertexGrid.InsertNextCell(aPolyVertex.GetCellType(),
-                                   aPolyVertex.GetPointIds())
+    if opacity is not None:
+        poly_actor.GetProperty().SetOpacity(opacity)
+    elif global_opacity >= 0:
+        poly_actor.GetProperty().SetOpacity(global_opacity)
+    poly_actor.GetProperty().SetPointSize(dot_size)
 
-    aPolyVertexGrid.SetPoints(polyVertexPoints)
-    aPolyVertexMapper = DataSetMapper()
-    aPolyVertexMapper.SetInputData(aPolyVertexGrid)
-    aPolyVertexActor = Actor()
-    aPolyVertexActor.SetMapper(aPolyVertexMapper)
+    return poly_actor
+
+
+def __color_check(dots, colors=None):
+    """
+    Returns a VTK scalar array containing colors information for each one of
+    the dots according to the policy defined by the parameter colors.
```

Comment:
This is a great generic function. Many actors could benefit from this, so, let's not limit it to the new dot actor only.